<h1 align="center" id="heading">ITBA</h1>
<h1 align="center" id="heading">Diplomatura en Cloud Data Engineering</h1>

## **Trabajo Práctico Intermedio Foundations**.
## Alumno: Jose Zerda

In [35]:
# Importamos Librerias:
import sqlalchemy
import psycopg2
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [36]:
# nos conectamos con la db:
%sql postgresql://postgres:postgres@app-postgres-db:5432/ds_salaries

### Probamos la conexión:

In [37]:
%%sql
select * from ds_salaries limit 6;

 * postgresql://postgres:***@app-postgres-db:5432/ds_salaries
6 rows affected.


[(1, '2023', 'SE', 'FT', 'Principal Data Scientist', 80000, 'EUR', 85847, 'ES', 100, 'ES', 'L'),
 (2, '2023', 'MI', 'CT', 'ML Engineer', 30000, 'USD', 30000, 'US', 100, 'US', 'S'),
 (3, '2023', 'MI', 'CT', 'ML Engineer', 25500, 'USD', 25500, 'US', 100, 'US', 'S'),
 (4, '2023', 'SE', 'FT', 'Data Scientist', 175000, 'USD', 175000, 'CA', 100, 'CA', 'M'),
 (5, '2023', 'SE', 'FT', 'Data Scientist', 120000, 'USD', 120000, 'CA', 100, 'CA', 'M'),
 (6, '2023', 'SE', 'FT', 'Applied Scientist', 222200, 'USD', 222200, 'US', 0, 'US', 'L')]

* Vemos que la conexión a nuestra base de datos funciona.

# **Sección consultas:**

## Consulta 1:
* ¿Cantidad de personas en las 12 posiciones laborales mas populosas de la encuesta?

In [38]:
%%sql

select job_title, count(*) AS cantidad
    from ds_salaries
    group by job_title
    order by cantidad DESC
    LIMIT 12;


 * postgresql://postgres:***@app-postgres-db:5432/ds_salaries
12 rows affected.


[('Data Engineer', 1040),
 ('Data Scientist', 840),
 ('Data Analyst', 612),
 ('Machine Learning Engineer', 289),
 ('Analytics Engineer', 103),
 ('Data Architect', 101),
 ('Research Scientist', 82),
 ('Applied Scientist', 58),
 ('Data Science Manager', 58),
 ('Research Engineer', 37),
 ('ML Engineer', 34),
 ('Data Manager', 29)]

### Respuesta 1:
* Las 5 posiciones mas numerosas son las siguientes:

1. Data Engineer con	1040
2. Data Scientist con	840
3. Data Analyst con 612
4. Machine Learning Engineer con	289
5. Analytics Engineer con	103

## Consulta 2:
* ¿De las 10 posiciones con mas presencia, cuales son los salarios promedio?

In [39]:
%%sql

SELECT job_title, AVG(salaryinusd) AS promedio, count(*) AS cantidad
FROM ds_salaries
GROUP BY job_title
ORDER BY cantidad DESC
LIMIT 12;



 * postgresql://postgres:***@app-postgres-db:5432/ds_salaries
12 rows affected.


[('Data Engineer', Decimal('142785.734615384615'), 1040),
 ('Data Scientist', Decimal('140869.765476190476'), 840),
 ('Data Analyst', Decimal('108716.227124183007'), 612),
 ('Machine Learning Engineer', Decimal('154690.726643598616'), 289),
 ('Analytics Engineer', Decimal('152368.631067961165'), 103),
 ('Data Architect', Decimal('161713.772277227723'), 101),
 ('Research Scientist', Decimal('161214.195121951220'), 82),
 ('Applied Scientist', Decimal('190264.482758620690'), 58),
 ('Data Science Manager', Decimal('191278.775862068966'), 58),
 ('Research Engineer', Decimal('163108.378378378378'), 37),
 ('ML Engineer', Decimal('158352.441176470588'), 34),
 ('Data Manager', Decimal('116613.344827586207'), 29)]

### Respuesta 2:
* **Data Science Manager** es la posicion mas remunerada de las primeras 10.
* **Data Analyst** es la posicion menos remunerada de las primeras 10.
* **Data Engineer y Data Scientist** las 2 posiciones mas numerosas tienen un ingreso de 140K U$s aprox.